### PR TITLE
LibCore: Port Environment to Windows and add most StandardPaths

### PR DESCRIPTION
**Pull request by stasoid**

### CODE REVIEW

File: Userland/Libraries/LibCore/StandardPaths.cpp [hunk 1]
```diff
@@ -229,6 +242,9 @@ ErrorOr<Vector<String>> StandardPaths::font_directories()
         // FIXME: We should be using the ASystemFontIterator NDK API here.
         // There is no guarantee that this will continue to exist on future versions of Android.
         "/system/fonts"_string,
+#    elif defined(AK_OS_WINDOWS)
+        "C:\\Windows\\Fonts"_string,
+        TRY(String::formatted(R"(C:\Users\{}\AppData\Local\Microsoft\Windows\Fonts)"sv, getenv("USERNAME"))),
```
Comment (konradekk) [inline Userland/Libraries/LibCore/StandardPaths.cpp:247]:
```suggestion
        TRY(String::formatted(R"({}\AppData\Local\Microsoft\Windows\Fonts)"sv, home_directory())),
```

or, probably even better,
```suggestion
        TRY(String::formatted(R"({}\Microsoft\Windows\Fonts)"sv, getenv("LOCALAPPDATA"))),
```

File: Userland/Libraries/LibCore/StandardPaths.cpp [hunk 2]
```diff
@@ -229,6 +242,9 @@ ErrorOr<Vector<String>> StandardPaths::font_directories()
         // FIXME: We should be using the ASystemFontIterator NDK API here.
         // There is no guarantee that this will continue to exist on future versions of Android.
         "/system/fonts"_string,
+#    elif defined(AK_OS_WINDOWS)
+        "C:\\Windows\\Fonts"_string,
```
Comment (konradekk) [inline Userland/Libraries/LibCore/StandardPaths.cpp:246]:
```suggestion
        TRY(String::formatted(R"({}\Fonts)"sv, getenv("WINDIR"))),
```

File: Userland/Libraries/LibCore/StandardPaths.cpp [hunk 3]
```diff
@@ -180,6 +187,8 @@ ErrorOr<ByteString> StandardPaths::runtime_directory()
 #elif defined(AK_OS_LINUX)
     auto uid = getuid();
     builder.appendff("/run/user/{}", uid);
+#elif defined(AK_OS_WINDOWS)
+    builder.appendff("{}", getenv("TEMP"));
```
Comment (konradekk) [inline Userland/Libraries/LibCore/StandardPaths.cpp:191]: maybe just
```suggestion
    builder.append(tempfile_directory());
```
?

**Review by konradekk — COMMENTED** [review]
some path remarks

File: Userland/Libraries/LibCore/StandardPaths.cpp [hunk 4]
```diff
@@ -36,19 +39,25 @@ static Optional<StringView> get_environment_if_not_empty(StringView name)
...
 }
 
 ByteString StandardPaths::desktop_directory()
 {
+#if !defined(AK_OS_WINDOWS)
     if (auto desktop_directory = get_environment_if_not_empty("XDG_DESKTOP_DIR"sv); desktop_directory.has_value())
         return LexicalPath::canonicalized_path(*desktop_directory);
+#endif
 
     StringBuilder builder;
```
Comment (konradekk) [inline Userland/Libraries/LibCore/StandardPaths.cpp:62]: probably not necessarily important at this moment, but…

you probably should reach to some windows api to get correct path for desktop directory; pretty much the same for documents, downloads, music, pictures, videos.

File: Userland/Libraries/LibCore/StandardPaths.cpp [hunk 5]
```diff
@@ -117,6 +136,10 @@ ByteString StandardPaths::videos_directory()
 
 ByteString StandardPaths::config_directory()
 {
+#ifdef AK_OS_WINDOWS
+    dbgln("Core::StandardPaths::config_directory() is not implemented");
+    VERIFY_NOT_REACHED();
+#endif
```
Comment (konradekk) [inline Userland/Libraries/LibCore/StandardPaths.cpp:142]: with the same into as above:

not sure what it used for, but if it is just to keep ladybird configuration, `%LocalAppData%/Ladybird` could be a decent storage path for that.

**Review by konradekk — COMMENTED** [review]
two additional notes

**Review by ADKaster — APPROVED** [review]

**Pull request merged**